src/main/java/core/basesyntax/DataConverterImpl.java

```diff
@@ -0,0 +1,34 @@
+package core.basesyntax;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    private static final int MAX_NUMBER_OF_PARTS = 3;
+
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> data) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+        for (String line : data) {
+            if (line.trim().isEmpty() || line.startsWith("type")) {
+                continue;
+            }
+            String[] parts = line.split(",");
+            if (parts.length != MAX_NUMBER_OF_PARTS) {
+                throw new RuntimeException("Invalid line format" + line);

```

Comment:
The error message for the RuntimeException should be more informative. Include information about what the expected format is. For example: 'Invalid line format: expected format <type,fruit,quantity>, got: ' + line